From 2a1ea4d02bc3fd90bf63099ba6752eb23808dff7 Mon Sep 17 00:00:00 2001
From: Ilia Kats <ilia-kats@gmx.net>
Date: Wed, 19 Apr 2023 00:06:42 +0200
Subject: [PATCH] KCM/mouse: enable compatibility with x11-libinput 1.3

The new version of libinput added an additional mouse acceleration
profile, so our checks were too restrictive and settings were not
applied.

BUG: 468217
---
 kcms/mouse/backends/x11/x11_libinput_dummydevice.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kcms/mouse/backends/x11/x11_libinput_dummydevice.cpp b/kcms/mouse/backends/x11/x11_libinput_dummydevice.cpp
index cb28980f2..e41731a2a 100644
--- a/kcms/mouse/backends/x11/x11_libinput_dummydevice.cpp
+++ b/kcms/mouse/backends/x11/x11_libinput_dummydevice.cpp
@@ -75,13 +75,13 @@ void valueWriterPart<bool>(bool val, Atom valAtom, Display *dpy)
             return;
         }
 
-        unsigned char sendVal[2] = {0};
+        unsigned char sendVal[3] = {0};
         if (num_items_return == 1) {
             sendVal[0] = val;
         } else {
             // Special case for acceleration profile.
             const Atom accel = XInternAtom(dpy, LIBINPUT_PROP_ACCEL_PROFILE_ENABLED, True);
-            if (num_items_return != 2 || valAtom != accel) {
+            if (num_items_return < 2 || num_items_return > 3 || valAtom != accel) {
                 return;
             }
             sendVal[val] = 1;
-- 
2.40.0

